src/wallet/gtest/test_wallet_cert.cpp

```diff
@@ -0,0 +1,124 @@
+#include <gtest/gtest.h>
+#include "gtest/tx_creation_utils.h"
+#include <boost/filesystem.hpp>
+
+#include <chainparams.h>
+#include <wallet/wallet.h>
+#include <wallet/walletdb.h>
+
+
+class CertInWalletTest : public ::testing::Test {
+public:
+    CertInWalletTest(): walletDbLocation(), pWallet(nullptr), pWalletDb(nullptr) {}
+    ~CertInWalletTest() = default;
+
+    void SetUp() override {
+        //Setup environment
+        SelectParams(CBaseChainParams::TESTNET);
+        boost::filesystem::path walletDbLocation = boost::filesystem::temp_directory_path() / boost::filesystem::unique_path();
+        boost::filesystem::create_directories(walletDbLocation);
+        mapArgs["-datadir"] = walletDbLocation.string();
+
+        //create wallet db
+        pWallet = new CWallet("wallet.dat");
+        try {  pWalletDb = new CWalletDB(pWallet->strWalletFile, "cr+"); }
+        catch(std::exception& e) {
+            ASSERT_TRUE(false)<<"Could not create tmp wallet db for reason "<<e.what();
+        }
+    };
+
+    void TearDown() override {
+        std::vector<std::shared_ptr<CWalletTransactionBase> > vWtx;
+        DBErrors nZapWalletRet = pWallet->ZapWalletTx(vWtx);
```

Comment:
nZapWalletRet is not used, should it be checked?

Reply:
the `ZapWalletTx `here makes sure that the wallet is duly cleaned-up and different UTs are well decoupled.
I would check `ZapWalletTx `return type with an EXPECT_TRUE(), in order to signal potential coupling of UTs, should `ZapWalletTx `fail.